### DISCUSSION TITLE
Request for help with custom router

### DISCUSSION THREAD

**Discussion by agorina** [fastapi/fastapi#7975] [Questions] [answered]
I have followed the instructions from https://fastapi.tiangolo.com/tutorial/custom-request-and-route/ and I do not see the custom router class being invoked.

Here is how my project is structured:

webapp.py:
```

"""A global API configuration
"""
app = FastAPI()
app.include_router(router.app_router, prefix="/api")
```

api.core.router.py:
```
""" Global API routers
"""
app_router = APIRouter()
app_router.route_class = TimedRoute
app_router.include_router(item_scorer.router)
```

api.endpoing.item_scorer:
```
router = APIRouter(route_class=TimedRoute)

@router.post("/score-item-response", response_model=MachineScore)
async def create_score_item_response(score_request: ItemScoreRequest) -> MachineScore:
    """An endpoint to score an item response
    """
    start_span({})
    return machine_scoring_engine.score_item(score_request)

```
TimedRoute (inside of app/api/core/sleuth.py)
```
class TimedRoute(APIRoute):
    def get_route_handler(self) -> Callable:
        original_route_handler = super().get_route_handler()

        async def custom_route_handler(request: Request) -> Response:
            logger.info("Test....")
            response: Response = await original_route_handler(request)
            logger.info("Test test...")
            return response

        return custom_route_handler
```
The `custom_route_handler` is never executed. Any help is very much appreciated.

Update:
I have also notices that `mypy` reports an error:
```app/api/core/sleuth.py:102: error: "get_route_handler" undefined in superclass``` 
This looks to me like a potential key to a problem?

**Comment by dmontagu** [comment]
What version of fastapi are you using?

**Comment by agorina** [comment]
0.38.1

**Comment by dmontagu** [comment, marked as answer]
This feature was added in 0.41 (current version is 0.42). Let us know if it doesn't work after you update!

**Comment by agorina** [comment]
Thank you, it works! I really appreciate a quick response!

**Comment by tiangolo** [comment]
Thanks for the help here @dmontagu ! :clap: :bow: 

Thanks for reporting back and closing the issue @agorina  :+1:
